### DISCUSSION TITLE
Component to present a timeline/chronological steps

### DISCUSSION THREAD

**Discussion by ghost** [twilio-labs/paste#3476] [Office hours] [answered]
### Background

We are building A/B experimentation into Segment's Broadcasts feature. This allows marketers to run experiments on different variants of their campaigns that are distributed to their test recipients, so they can validate which variant performs the best.

For this functionality, we need a component that allows for users to preview a sequence of experiment steps (e.g. start test, run test, etc) and their relevant dates. _When I am reviewing my experiment in Broadcasts, I want to preview a flow of my experiment, so that I can be confident in each step of my experiment and ensure its success._

The current proposal for this preview of steps is adapting the visual picker component for this use case. Similar usage was was seen in some of the [paste migration work for the delivery overview](https://www.figma.com/file/oeUbF1qpn2FUsmXwiTsHmN/Segment-App%3A-Paste-Migration?type=design&node-id=660%3A228700&mode=design&t=V8SOmZ0BWhBeBVfz-1). 

<img width="1002" alt="Screenshot 2023-09-06 at 4 48 51 PM" src="https://github.com/twilio-labs/paste/assets/103461187/7e2210d1-e984-433b-910b-ab3b3d65c4f1">

### Feedback

Is this an advisable usage of this component? Is there another approach that might better support this use case? 

In this proposal, are the icons, background, and border color appropriate to convey the status of each step? 

### Stage of design

Exploration, high fidelity

### Research

[Experimentation foundational research](https://docs.google.com/document/d/15BP0Y9y1lMg2u48oETTCSv-qRQe4WTo4Jj4Tz9n3Ods/edit#heading=h.ttue4a2cy8z0)

### Relevant links

[Figma explorations](https://www.figma.com/file/kp9Jout2yMO6IKE6MbfpVG/%5BDesign%5D-Experimentation-%E2%80%93-Broadcasts-and-messages?type=design&node-id=1023%3A407842&mode=design&t=Oea1eVWCyjQPAGK9-1)

### Are you looking for visual design support?

- [ ] Yes

**Comment by ghost** [comment]
Following up from our office hours discussion on 9/7, it was advised that the progress steps component should be leveraged for our use case. We have since updated the [design explorations](https://www.figma.com/file/kp9Jout2yMO6IKE6MbfpVG/%5BDesign%5D-Experimentation-%E2%80%93-Broadcasts-and-messages?type=design&node-id=1155%3A432312&mode=design&t=l3ryvDVKQq8q1A4p-1), and @msaunders-twilio has completed a spike using progress steps.
 
![Screenshot 2023-09-08 at 10 59 23 AM](https://github.com/twilio-labs/paste/assets/103461187/28a83ad8-3c8b-4e5b-9675-c42a79bb9e86)

We'd appreciate any feedback around best presenting the step metadata (i.e. dates, test percentages) under each step. How should we consider alignment of icons to the step names here?

**Reply by serifluous** [reply]
Hey @chrischan-twilio ! Did you get a chance to do any explorations beyond this, like some of the suggestions @nkrantz had in discussion #3389 ?

Seeing these explorations with Progress Steps, it feels like it might be stretched beyond what it's intended to do. Thoughts?

**Comment by serifluous** [comment, marked as answer]
Thanks for coming to office hours! Here's the recording: [https://twilio.zoom.us/rec/share/APhhuOP0V-V4XCIguVpqMhuQfppNuJbhTz4HGVu9FCBhXRP_FOCsZV9QVRQIaOk.14YO9Yyx8oQtZfI_](https://twilio.zoom.us/rec/share/APhhuOP0V-V4XCIguVpqMhuQfppNuJbhTz4HGVu9FCBhXRP_FOCsZV9QVRQIaOk.14YO9Yyx8oQtZfI_ )You can find the password in #help-design-system
